compiler/modules/front.syntax/ast.bal

```diff
@@ -351,7 +351,7 @@ public type FieldAccessExpr record {|
 public type FieldAccessLExpr record {|
     *PositionFields;
     Position opPos;
-    VarRefExpr container;
+    VarRefExpr|FieldAccessLExpr container;
```

Comment:
This does not restrict things in the right way. The thing we cannot suppport is member access as an *intermediate* step, because that is what uses a filling read. So `x.y.z[0]` is fine but `x[0].y.z` isn't.